lib/active_record_host_pool/connection_adapter_mixin.rb

```diff
@@ -87,9 +87,17 @@ module ConnectionAdapters
     class ConnectionHandler
 
       if ActiveRecord::VERSION::MAJOR >= 5
-
-        def establish_connection(spec)
-          owner_to_pool[spec.name] = ActiveRecordHostPool::PoolProxy.new(spec)
+        if ActiveRecord::VERSION::MINOR == 0
+          def establish_connection(spec)
+            owner_to_pool[spec.name] = ActiveRecordHostPool::PoolProxy.new(spec)
+          end
+        else
+          def establish_connection(spec)
```

Comment:
This method now receives a hash so we need to get the specification for the rest to work.